src/data/Data.js

```diff
@@ -643,25 +644,39 @@ export const WorkspaceData = _.flow(
       }
     };
 
-    const loadWdsUrl = useCallback((workspaceId) => {
-      return Ajax()
-        .Apps.listAppsV2(workspaceId)
-        .then((apps) => {
-          const foundApp = resolveWdsApp(apps);
-          if (foundApp?.status === appStatuses.running.status) {
-            const url = foundApp.proxyUrls.wds;
-            setWdsProxyUrl({ status: 'Ready', state: url });
-            return url;
-          }
-          if (foundApp?.status === appStatuses.error.status) {
-            setWdsProxyUrl({ status: 'Error', state: 'WDS app is in ERROR state' });
-          }
-        })
-        .catch((error) => {
-          setWdsProxyUrl({ status: 'Error', state: 'Error resolving WDS app' });
-          reportError('Error resolving WDS app', error);
-        });
-    }, []);
+    const loadWdsApp = useCallback(
+      (workspaceId) => {
+        return Ajax()
+          .Apps.listAppsV2(workspaceId)
+          .then((apps) => {
+            const foundApp = resolveWdsApp(apps);
+            switch (foundApp?.status) {
+              case appStatuses.provisioning.status:
+              case appStatuses.updating.status:
+                setWdsApp({ status: 'Loading', state: foundApp });
+                return;
+              case appStatuses.running.status:
+                setWdsApp({ status: 'Ready', state: foundApp });
+                return wdsApp;
+              case appStatuses.error.status:
+                setWdsApp({ status: 'Error', state: foundApp });
+                return undefined;
+              default:
+                if (foundApp?.status) {
+                  // eslint-disable-next-line no-console
+                  console.log(`Unhandled state [${foundApp?.status} while polling WDS`);
+                }
+
+                return undefined;
+            }
+          })
+          .catch((error) => {
+            setWdsApp({ status: 'Error', state: 'Error resolving WDS app' });
+            reportError('Error resolving WDS app', error);
+          });
+      },
+      [wdsApp]
```

Comment:
Does this need to depend on `wdsApp`?

Reply:
Nope, that was leftover from attempts at addressing the TODO I left earlier on.  Thanks for noticing!  I removed this in the followup commit.